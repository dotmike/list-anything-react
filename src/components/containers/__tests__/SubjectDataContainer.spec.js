import configureStore from "redux-mock-store";

import ConnectedSubjectDataContainer, {
  UnconnectedSubjectDataContainer
} from "../SubjectDataContainer";

const setup = (propOverrides = {}) => {
  const props = {
    menuVisible: false,
    onMenuClick: jest.fn(),
    children: "Testando...",
    activeTab: "",
    tabs: [],
    ...propOverrides
  };

  return {
    props,
    connectedWrapperShallow: wrapperShallow(
      ConnectedSubjectDataContainer,
      props,
      true
    ),
    wrapperShallow: wrapperShallow(UnconnectedSubjectDataContainer, props, true)
  };
};

const testData = {};
testData.tabs = [{ _id: "1", description: "Test" }];
testData.activeTab = testData.tabs[0]._id;
testData.tabid = testData.activeTab;

describe("ConnectedSubjectDataContainer", () => {
  const mockStore = configureStore();
  const state = {};
  it("renders correctly", () => {
    const { connectedWrapperShallow: wrapper } = setup({
      store: mockStore(state)
    });
    expect(wrapper()).toMatchSnapshot();
  });
});

describe("UnconnectedSubjectDataContainer", () => {
  it("renders correctly when `tabs` is empty", () => {
    const { wrapperShallow: wrapper } = setup({ tabs: [] });
    expect(wrapper()).toMatchSnapshot();
  });

  it("renders correctly when `tabs` is not empty", () => {
    const { wrapperShallow: wrapper } = setup({
      tabs: testData.tabs
    });
    expect(wrapper()).toMatchSnapshot();
  });

  it("renders correctly when `menuVisible` is false", () => {
    const { wrapperShallow: wrapper } = setup({
      menuVisible: false
    });
    expect(wrapper()).toMatchSnapshot();
  });

  it("renders correctly when `menuVisible` is true", () => {
    const { wrapperShallow: wrapper } = setup({
      menuVisible: true
    });
    expect(wrapper()).toMatchSnapshot();
  });

  it("renders correctly when `activeTab` is set", () => {
    const { wrapperShallow: wrapper } = setup({
      tabs: testData.tabs,
      activeTab: testData.activeTab
    });
    expect(wrapper()).toMatchSnapshot();
  });

  it("calls `onMenuClick` when clicking on a `Sidemenu Item`", () => {
    const { wrapperShallow: wrapper, props } = setup({
      tabs: testData.tabs
    });

    wrapper()
      .find({ tabid: testData.tabid })
      .simulate("click");
    expect(props.onMenuClick).toHaveBeenCalled();
  });
});
