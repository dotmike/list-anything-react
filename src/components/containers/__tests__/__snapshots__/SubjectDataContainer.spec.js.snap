// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedSubjectDataContainer renders correctly 1`] = `
<SubjectDataContainer
  activeTab=""
  dispatch={[Function]}
  menuVisible={false}
  onMenuClick={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  tabs={Array []}
>
  Testando...
</SubjectDataContainer>
`;

exports[`UnconnectedSubjectDataContainer renders correctly when \`activeTab\` is set 1`] = `
<SidebarPushable
  as={[Function]}
  attached="bottom"
  className="subject-data-container"
>
  <Sidebar
    animation="overlay"
    as={[Function]}
    direction="left"
    icon="labeled"
    inverted={true}
    vertical={true}
    visible={false}
    width="thin"
  >
    <MenuItem
      active={true}
      key="1"
      onClick={[Function]}
      tabid="1"
    >
      Test
    </MenuItem>
  </Sidebar>
  <SidebarPusher
    dimmed={false}
    style={
      Object {
        "overflowY": "auto",
      }
    }
  >
    Testando...
  </SidebarPusher>
</SidebarPushable>
`;

exports[`UnconnectedSubjectDataContainer renders correctly when \`menuVisible\` is false 1`] = `
<SidebarPushable
  as={[Function]}
  attached="bottom"
  className="subject-data-container"
>
  <Sidebar
    animation="overlay"
    as={[Function]}
    direction="left"
    icon="labeled"
    inverted={true}
    vertical={true}
    visible={false}
    width="thin"
  />
  <SidebarPusher
    dimmed={false}
    style={
      Object {
        "overflowY": "auto",
      }
    }
  >
    Testando...
  </SidebarPusher>
</SidebarPushable>
`;

exports[`UnconnectedSubjectDataContainer renders correctly when \`menuVisible\` is true 1`] = `
<SidebarPushable
  as={[Function]}
  attached="bottom"
  className="subject-data-container"
>
  <Sidebar
    animation="overlay"
    as={[Function]}
    direction="left"
    icon="labeled"
    inverted={true}
    vertical={true}
    visible={true}
    width="thin"
  />
  <SidebarPusher
    dimmed={true}
    style={
      Object {
        "overflowY": "auto",
      }
    }
  >
    Testando...
  </SidebarPusher>
</SidebarPushable>
`;

exports[`UnconnectedSubjectDataContainer renders correctly when \`tabs\` is empty 1`] = `
<SidebarPushable
  as={[Function]}
  attached="bottom"
  className="subject-data-container"
>
  <Sidebar
    animation="overlay"
    as={[Function]}
    direction="left"
    icon="labeled"
    inverted={true}
    vertical={true}
    visible={false}
    width="thin"
  />
  <SidebarPusher
    dimmed={false}
    style={
      Object {
        "overflowY": "auto",
      }
    }
  >
    Testando...
  </SidebarPusher>
</SidebarPushable>
`;

exports[`UnconnectedSubjectDataContainer renders correctly when \`tabs\` is not empty 1`] = `
<SidebarPushable
  as={[Function]}
  attached="bottom"
  className="subject-data-container"
>
  <Sidebar
    animation="overlay"
    as={[Function]}
    direction="left"
    icon="labeled"
    inverted={true}
    vertical={true}
    visible={false}
    width="thin"
  >
    <MenuItem
      active={false}
      key="1"
      onClick={[Function]}
      tabid="1"
    >
      Test
    </MenuItem>
  </Sidebar>
  <SidebarPusher
    dimmed={false}
    style={
      Object {
        "overflowY": "auto",
      }
    }
  >
    Testando...
  </SidebarPusher>
</SidebarPushable>
`;
